### PR TITLE
Fixed issue #L41-T1509: Numeric questions prevent statistics from generating and display false data

### PR DESCRIPTION
**Pull request by ptelu**

### CODE REVIEW

File: application/helpers/common_helper.php [hunk 1]
```diff
@@ -5087,3 +5087,23 @@ function recursive_preg_replace($pattern, $replacement, $subject, $limit = -1, &
     }
     return $result;
 }
+
+/**
+ * function to calculate the standard deviation of an array
+ * @var array $array The array to calculate the standard deviation for
+ * @return float
+ */
+function standard_deviation(array $array): float
```
Comment (olleharstedt) [inline application/helpers/common_helper.php:5096]: All new functions should be camelCase

File: application/helpers/common_helper.php [hunk 2]
```diff
@@ -5087,3 +5087,23 @@ function recursive_preg_replace($pattern, $replacement, $subject, $limit = -1, &
     }
     return $result;
 }
+
+/**
+ * function to calculate the standard deviation of an array
+ * @var array $array The array to calculate the standard deviation for
```
Comment (olleharstedt) [inline application/helpers/common_helper.php:5093]: @var should be @param 

If it's an array of floats, can you document it as type `float[]`, please?

File: application/helpers/common_helper.php (hunk 1 above)
Comment (olleharstedt) [reply application/helpers/common_helper.php:5096]: $array can be named $numbers or $floats if it's an array of numbers.

File: application/helpers/common_helper.php [hunk 3]
```diff
@@ -5087,3 +5087,23 @@ function recursive_preg_replace($pattern, $replacement, $subject, $limit = -1, &
     }
     return $result;
 }
+
+/**
+ * function to calculate the standard deviation of an array
```
Comment (olleharstedt) [inline application/helpers/common_helper.php:5092]: Maybe rephrase as

"Returns the standard deviation of $numbers" or "Calculates the standard deviation of $numbers"

File: application/helpers/admin/statistics_helper.php [hunk 4]
```diff
@@ -4115,50 +4099,53 @@ public function generate_statistics($surveyid, $allfields, $q2show = 'all', $use
...
         static $sid = null;
         static $recordCount = 0;
         static $field = null;
         static $allRows = null;
 
-        if ($surveyid !== $sid || $fieldname !== $field) {
+        $criteria = new CDbCriteria();
+        if ($fielddata['sid'] !== $sid || $fielddata['fieldname'] !== $field) {
             //get data
-            $query = " FROM {{survey_$surveyid}} WHERE " . Yii::app()->db->quoteColumnName($fieldname) . " IS NOT null";
```
Comment (olleharstedt) [inline application/helpers/admin/statistics_helper.php:4133]: +1 for getting rid of raw SQL!

**Comment by olleharstedt** [comment]
Hm, I already did review. @ptelu Can you apply the suggested changes and then merge?

File: application/helpers/admin/statistics_helper.php (hunk 4 above)
Comment (Shnoulle) [reply application/helpers/admin/statistics_helper.php:4133]: +2
